commit 157b283fdbcdb279f15a85bcccba835e05e44ac5
Author: dann frazier <dannf@hp.com>
Date:   Mon Feb 15 12:33:49 2010 -0700

    [Adjusted to apply to Debian's 2.6.24]
    
    commit dfe195fb79e88c334481f1362fef52f6d2e30b2d
    Author: Benny Halevy <bhalevy@panasas.com>
    Date:   Tue Aug 5 13:01:41 2008 -0700
    
        mm: fix uninitialized variables for find_vma_prepare callers

diff --git a/mm/mmap.c b/mm/mmap.c
index 147cf98..8729fb2 100644
--- a/mm/mmap.c
+++ b/mm/mmap.c
@@ -352,7 +352,7 @@ find_vma_prepare(struct mm_struct *mm, unsigned long addr,
 		if (vma_tmp->vm_end > addr) {
 			vma = vma_tmp;
 			if (vma_tmp->vm_start <= addr)
-				return vma;
+				break;
 			__rb_link = &__rb_parent->rb_left;
 		} else {
 			rb_prev = __rb_parent;
